src/app/snapshot/snapshots-page/snapshot-sidebar/snapshot-sidebar.container.ts

```diff
@@ -1,30 +1,59 @@
 import { Component, OnInit } from '@angular/core';
 import { ActivatedRoute } from '@angular/router';
 import { select, Store } from '@ngrx/store';
-import { State } from '../../../reducers';
+import { combineLatest, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
 
+import { State } from '../../../reducers';
 import * as fromSnapshots from '../../../reducers/snapshots/redux/snapshot.reducers';
+import * as fromVm from '../../../reducers/vm/redux/vm.reducers';
 import * as fromVolumes from '../../../reducers/volumes/redux/volumes.reducers';
-import * as snapshotActions from '../../../reducers/snapshots/redux/snapshot.actions';
+import { vmSnapshotsSelectors } from '../../../root-store/server-data/vm-snapshots';
+import { Snapshot } from '../../../shared/models';
+import { VmSnapshotSidebarViewModel } from '../../models/vm-snapshot-sidebar.view-model';
+import { snapshotPageActions, snapshotPageSelectors } from '../../store';
+import { SnapshotPageViewMode } from '../../types';
 
 @Component({
   selector: 'cs-snapshot-sidebar-container',
   template: `
-    <cs-snapshot-sidebar
-      [snapshot]="snapshot$ | async"
-      [volumes]="volumes$ | async"
-      [isLoading]="isLoading$ | async"
-    ></cs-snapshot-sidebar>`,
+    <ng-container *loading="isLoading$ | async">
+      <ng-container *ngIf="snapshot$ | async as snapshot" [ngSwitch]="viewMode$ | async">
+        <cs-snapshot-sidebar
+          *ngSwitchCase="viewModeType.Volume"
+          [snapshot]="snapshot"
+          [volumes]="volumes$ | async"
+          [isLoading]="isLoading$ | async"
+        ></cs-snapshot-sidebar>
+        <cs-vm-snapshot-sidebar
+          *ngSwitchCase="viewModeType.VM"
+          [vmSnapshot]="snapshot"
+        ></cs-vm-snapshot-sidebar>
+      </ng-container>
+    </ng-container>
+  `,
 })
 export class SnapshotSidebarContainerComponent implements OnInit {
-  readonly snapshot$ = this.store.pipe(select(fromSnapshots.getSelectedSnapshot));
+  readonly snapshot$: Observable<Snapshot | VmSnapshotSidebarViewModel> = this.store.pipe(
+    select(snapshotPageSelectors.getSelectedSnapshotForSidebar),
+  );
+  readonly viewMode$: Observable<SnapshotPageViewMode> = this.store.pipe(
+    select(snapshotPageSelectors.getViewMode),
+  );
   readonly volumes$ = this.store.pipe(select(fromVolumes.selectEntities));
-  readonly isLoading$ = this.store.pipe(select(fromSnapshots.isLoading));
+  readonly isLoading$ = combineLatest(
+    this.store.pipe(select(fromSnapshots.isLoading)),
+    this.store.pipe(select(vmSnapshotsSelectors.getIsLoading)),
+    // you need to wait for virtual machines to load for vmSnapshot selectors
```

Comment:
Maybe you should put the `fromVm.isLoading` check directly into vmSnapshotsSelectors.getIsLoading?